interface/src/app/sample/services.js

```diff
@@ -0,0 +1,37 @@
+angular.module('adage.sample.services', [
+  'ngResource',
+  'adage.utils'
+])
+
+.factory('Sample', ['$resource', '$http', 'ApiBasePath',
+  function($resource, $http, ApiBasePath) {
+    var Sample = $resource(
+      ApiBasePath + 'sample/:id/',
+      // TODO need to add logic for handling pagination of results.
+      // then, can change "limit" below to something sensible
+      {id: '@id', limit: 0},
+      // Angular expects a query service to give only a list but our Tastypie
+      // interface wraps the response list with pagination so we tell Angular to
+      // expect an object instead via isArray: false
+      {
+        'get': {
+          method: 'GET',
+          isArray: false
+        },
+        'getExperiments': {
+          url: ApiBasePath + 'sample/:id/get_experiments/',
+          method: 'GET',
+          isArray: true
+        }
+      }
+    );
+    Sample.getUri = function(uri) {
+      // We can't easily get Tastypie-supplied resource_uris via $resource, so
+      // this uses $http directly and returns the HttpPromise
+      return $http({url: uri, method: 'GET'});
+    };
+    return Sample;
+  }
+])
+
```

Comment:
I wonder whether the definition of this service can be simplified into:
```
.factory('Sample', ['$resource', 'ApiBasePath',
  function($resource, ApiBasePath) {
    var Sample = $resource(
      ApiBasePath + 'sample/:id/',
      {limit: 0},
      {
        'getExperiments': {
          url: ApiBasePath + 'sample/:id/get_experiments/',
          method: 'GET',
          isArray: true
        }
      }
    );
    Sample.getUri = function(uri) {
      return $resource(uri).get().$promise;
    };
    return Sample;
  }
])
```
(I removed the comments to make the code easier to read. Of course you can keep them.)

In getUri() method, `$http` dependency is replaced by `$promise` property. I also feel that the method name `getUri` is misleading, since it actually gets a promise.

By the way, getUri() method seems to be unrelated to Sample at all. Why not create another service like this:
```
.factory('getPromise', ['$resource', 
  function($resource) {
    var fn =  function(uri) {
      return $resource(uri).get().$promise;
    };
    return fn;
  }
]);
```